### PR TITLE
Fix minor js chapters issues

### LINKED ISSUE #132: Chapter on writing testable JavaScript implictly uses jQuery
For eg: https://github.com/se-edu/learningresources/blob/master/contents/javascript/javascript.md#avoid-coupling-with-selectors

The first few examples all assume the usage of jQuery ($, jquery selectors, $.ajax etc), but is not stated anywhere before them.

Should we try to change these examples so that they don't require any extra libraries, and use vanilla js instead so that they can be applicable to anyone who uses js?

### LINKED ISSUE #133: Improve default parameters example
https://github.com/se-edu/learningresources/blob/master/contents/javascript/javascript.md#writing-reusable-javascript

The example:
```js
// es5 syntax
function createPopUp(title, content, status, optionals) {
    var headerColor = optionals.headerColor || 'default';
    var bodyColor = optionals.bodyColor || 'default';
    ...
}
// es6 syntax with destructuring and default parameters
function createPopUp(title, content, status, optionals) {
    const { headerColor = 'default', bodyColor = 'default' } = optionals;
    ...
}
```

the destructuring is more declatively done like:
```
// es6 syntax with destructuring and default parameters
function createPopUp(title, content, status, {headerColor = 'default', bodyColor = 'default'}) {
    ...
}
```

### LINKED ISSUE #134: Incorrect code in javascript.md#writing-reusable-javascript
Quote from chapter:
https://github.com/se-edu/learningresources/blob/master/contents/javascript/javascript.md#writing-reusable-javascript
> Method chaining is syntax such as `array.concat([1, 2]).push(1).filter(isEven)`

image

However, `Array.prototype.push` doesn't return an array, the above code doesn't do what it intends to do.

### LINKED ISSUE #135: Use of word 'module' in javascript/javascript-modules#module-pattern confusing
Link:

https://deploy-preview-127--learningresources.netlify.com/contents/javascript/javascript-modules#module-pattern

image

JavaScript has modules already (as described in the section directly above). While the above code does make use of the module pattern (wrapping all related code inside one global), calling that a module just adds to the confusion with the already two different module systems: ES6 modules and CommonJS modules.

I propose we change rename module pattern to ['namespace pattern'](https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html)
to better reflect what is going on: the reduction of global variables pollution by only creating one global variable, with all other functions and variables inside that object.

### PR DESCRIPTION
**Pull request by openorclose**
Fixes #135 
Fixes #134 
Fixes #133 
Fixes #132 

A bit iffy on the module pattern renaming, any thoughts welcome.

### CODE REVIEW

File: contents/javascript/Javascript-modules.md [hunk 1]
```diff
@@ -71,13 +71,13 @@ ES6 modules supports advanced features such as _[asynchronous loading](http://ex
 
 Due to the very recent adoption of ES6 modules by browsers and there is still some browsers that do not support it, you might not be able to use ES6 modules right away in those unsupported browsers.
 
-There are 2 workarounds for this issue. You can use _[transpiler](https://scotch.io/tutorials/javascript-transpilers-what-they-are-why-we-need-them)_ such as _[Babel](https://babeljs.io/)_ and _[bundler](https://medium.com/@gimenete/how-javascript-bundlers-work-1fc0d0caf2da)_ such as _[Webpack](https://webpack.js.org/)_ to serve your application to those unsupported browsers or use one of the other two approaches mentioned below.
+There are 2 workarounds for this issue. You can use a _[transpiler](https://scotch.io/tutorials/javascript-transpilers-what-they-are-why-we-need-them)_ such as _[Babel](https://babeljs.io/)_ and a _[bundler](https://medium.com/@gimenete/how-javascript-bundlers-work-1fc0d0caf2da)_ such as _[Webpack](https://webpack.js.org/)_ to serve your application to those unsupported browsers or use one of the other two approaches mentioned below.
```
Comment (Tejas2805) [inline contents/javascript/Javascript-modules.md]: Since you have written - there are 2 workarounds for this issue - Shouldn't it `babel` **or** `bundler`

**Review by Tejas2805 — CHANGES_REQUESTED** [review]

File: contents/javascript/Javascript-modules.md (hunk 1 above)
Comment (openorclose) [reply contents/javascript/Javascript-modules.md]: Right, I'll change it

File: contents/javascript/Javascript-modules.md [hunk 2]
```diff
@@ -71,13 +71,13 @@ ES6 modules supports advanced features such as _[asynchronous loading](http://ex
 
 Due to the very recent adoption of ES6 modules by browsers and there is still some browsers that do not support it, you might not be able to use ES6 modules right away in those unsupported browsers.
```
Comment (jamessspanggg) [inline contents/javascript/Javascript-modules.md]: Could you also modify the line here to 'manageable'?

https://github.com/se-edu/learningresources/blame/93c24e7d4da32a78f80091cf05bfc3fc195792a3/contents/javascript/Javascript-modules.md#L30

Comment (jamessspanggg) [reply contents/javascript/Javascript-modules.md]: rely on a -> relies on a 
https://github.com/se-edu/learningresources/blame/93c24e7d4da32a78f80091cf05bfc3fc195792a3/contents/javascript/Javascript-modules.md#L39

Comment (jamessspanggg) [reply contents/javascript/Javascript-modules.md]: there is -> there are

https://github.com/se-edu/learningresources/blame/93c24e7d4da32a78f80091cf05bfc3fc195792a3/contents/javascript/Javascript-modules.md#L49

https://github.com/se-edu/learningresources/blame/93c24e7d4da32a78f80091cf05bfc3fc195792a3/contents/javascript/Javascript-modules.md#L72

**Review by jamessspanggg — CHANGES_REQUESTED** [review]
Some nits

File: contents/javascript/Javascript-modules.md (hunk 2 above)
Comment (openorclose) [reply contents/javascript/Javascript-modules.md]: Thanks! I initially only meant to change the outdated or incorrect js stuff, I'll go ahead and change these other minor issues too

Comment (openorclose) [reply contents/javascript/Javascript-modules.md]: Done!

**Review by jamessspanggg — APPROVED** [review]
LGTM

**Comment by alyip98** [comment]
LGTM!

**Comment by openorclose** [comment]
@damithc is this ready to be merged?

**Pull request merged**